cirq-core/cirq/ops/projector.py

```diff
@@ -0,0 +1,161 @@
+import itertools
+from typing import (
+    Any,
+    Dict,
+    Iterable,
+    List,
+    Mapping,
+    Optional,
+    Union,
+)
+
+import numpy as np
+from scipy.sparse import coo_matrix
+
+from cirq import value
+from cirq.ops import raw_types
+
+
+def check_qids_dimension(qids):
+    """A utility to check that we only have Qubits."""
+    for qid in qids:
+        if qid.dimension != 2:
+            raise ValueError(f"Only qubits are supported, but {qid} has dimension {qid.dimension}")
+
+
+@value.value_equality
+class ProjectorString:
+    def __init__(
+        self,
+        projector_dict: Dict[raw_types.Qid, int],
+        coefficient: Union[int, float, complex] = 1,
+    ):
+        """Contructor for ProjectorString
+
+        Args:
+            projector_dict: a dictionary of Qbit to an integer specifying which vector to project
```

Comment:
Nit: "projector_dict: A python dictionary mapping from `cirq.Qid` to integers. A key value pair represents the desired computational basis state for that qubit."

Reply:
Done.